certs/service.go

```diff
@@ -127,80 +100,76 @@ func (cs *certsService) RevokeCert(ctx context.Context, token, thingID string) (
-	u, err := cs.auth.Identify(ctx, &magistrala.IdentityReq{Token: token})
+func (cs *certsService) ListCerts(ctx context.Context, token, thingID string, offset, limit uint64) (sdk.CertificatePage, error) {
+	_, err := cs.auth.Identify(ctx, &magistrala.IdentityReq{Token: token})
 	if err != nil {
-		return Page{}, errors.Wrap(svcerr.ErrAuthentication, err)
+		return sdk.CertificatePage{}, errors.Wrap(svcerr.ErrAuthentication, err)
 	}
 
-	cp, err := cs.certsRepo.RetrieveByThing(ctx, u.GetId(), thingID, offset, limit)
+	cp, err := cs.pki.ListCerts(sdk.PageMetadata{Offset: offset, Limit: limit, EntityID: thingID})
 	if err != nil {
-		return Page{}, errors.Wrap(svcerr.ErrViewEntity, err)
-	}
-
-	for i, cert := range cp.Certs {
-		vcert, err := cs.pki.Read(cert.Serial)
-		if err != nil {
-			return Page{}, errors.Wrap(ErrFailedReadFromPKI, err)
-		}
-		cp.Certs[i].ClientCert = vcert.ClientCert
-		cp.Certs[i].ClientKey = vcert.ClientKey
+		return sdk.CertificatePage{}, errors.Wrap(svcerr.ErrViewEntity, err)
 	}
 
 	return cp, nil
 }
 
-func (cs *certsService) ListSerials(ctx context.Context, token, thingID string, offset, limit uint64) (Page, error) {
-	u, err := cs.auth.Identify(ctx, &magistrala.IdentityReq{Token: token})
+func (cs *certsService) ListSerials(ctx context.Context, token, thingID string, offset, limit uint64) (sdk.CertificatePage, error) {
+	_, err := cs.auth.Identify(ctx, &magistrala.IdentityReq{Token: token})
 	if err != nil {
-		return Page{}, errors.Wrap(svcerr.ErrAuthentication, err)
+		return sdk.CertificatePage{}, errors.Wrap(svcerr.ErrAuthentication, err)
 	}
 
-	return cs.certsRepo.RetrieveByThing(ctx, u.GetId(), thingID, offset, limit)
-}
-
-func (cs *certsService) ViewCert(ctx context.Context, token, serialID string) (Cert, error) {
-	u, err := cs.auth.Identify(ctx, &magistrala.IdentityReq{Token: token})
+	cp, err := cs.pki.ListCerts(sdk.PageMetadata{Offset: offset, Limit: limit, EntityID: thingID})
 	if err != nil {
-		return Cert{}, errors.Wrap(svcerr.ErrAuthentication, err)
+		return sdk.CertificatePage{}, errors.Wrap(svcerr.ErrViewEntity, err)
+	}
+
+	var certs []sdk.Certificate
+	for _, c := range cp.Certificates {
+		if !c.Revoked {
+			certs = append(certs, sdk.Certificate{
+				SerialNumber: c.SerialNumber,
+				EntityID:     c.EntityID,
+				ExpiryTime:   c.ExpiryTime,
+				Revoked:      c.Revoked,
+			})
+		}
 	}
+	return sdk.CertificatePage{
+		Offset:       cp.Offset,
+		Limit:        cp.Limit,
+		Total:        uint64(len(certs)),
+		Certificates: certs,
+	}, nil
+}
 
-	cert, err := cs.certsRepo.RetrieveBySerial(ctx, u.GetId(), serialID)
+func (cs *certsService) ViewCert(ctx context.Context, token, serialID string) (sdk.Certificate, error) {
+	_, err := cs.auth.Identify(ctx, &magistrala.IdentityReq{Token: token})
 	if err != nil {
-		return Cert{}, errors.Wrap(svcerr.ErrViewEntity, err)
+		return sdk.Certificate{}, errors.Wrap(svcerr.ErrAuthentication, err)
```

Comment:
Identify returns an svcerr.ErrAuthentication, so this is repeated.  You can revert